guidelines/swiftui/guidelines.md

```diff
@@ -0,0 +1,173 @@
+- Avoid UIKit unless SwiftUI cannot support a use-case (justify and document exception)
+- Modularize by feature, not by type (views, models, services grouped together per feature)
+
+---
+
+## Architecture Standards (MVVM)
+- Flow: Views → ViewModels → Service / API Client
+- API Client → Network Layer (URLSession or 3rd-party lib)
+- Never expose raw models to the View layer
+- Each ViewModel owns a protocol-conforming Service instance injected via initializer
+- Use `Task {}` blocks for async loading in ViewModels, with proper cancellation handling
+
+### Example
+
+```swift
+protocol ExampleServiceProtocol {
+    func fetchDailyQuote() async throws -> String
+}
+
+final class ExampleViewModel: ObservableObject {
+    @Published var quote: String = ""
+    private let service: ExampleServiceProtocol
+
+    init(service: ExampleServiceProtocol) {
+        self.service = service
+    }
+
+    func load() async {
+        do {
+            quote = try await service.fetchDailyQuote()
+        } catch {
+            // Handle gracefully
+        }
+    }
+}
+```
+
+## Architecture Standards (MVVM + Repository)
+- View → ViewModel
+- ViewModel → Repository (abstracts data source)
+- Repository → Service / API Client
+- Each ViewModel owns a protocol-conforming Service instance injected via initializer
+- In case you neen a Repository, Each ViewModel owns a protocol-conforming Repository instance, injected via initializer.
+  Each Repository owns one or more protocol-conforming Service/API clients, injected via initializer.
+
+### Example
+
+```swift
+final class ProfileViewModel: ObservableObject {
+    private let repository: ProfileRepository
+
+    init(repository: ProfileRepository) {
+        self.repository = repository
+    }
+
+    func loadProfile() {
+        Task {
+            do {
+                let profile = try await repository.fetchUserProfile()
+                // map to ViewModel state here
+            } catch {
+                // handle error
+            }
+        }
+    }
+}
+```
+---
+
+## Swift Style Guide
+- Use `camelCase` for properties, methods
+- Use `PascalCase` for types, structs, enums
+- Use `let` over `var` wherever possible
+- Max line length: 120 characters
+- Use `guard` for early exits
+- Avoid magic numbers; use constants/enums
+- Use constants for spaces and colors
+- Use trailing closures
+- Add inline comments only when logic is non-obvious
+
+---
+
+## Async & Concurrency
+- Use `async throws` for all network calls and async operations
+- Cancel outdated tasks with `@MainActor` or `Task.cancel()` as needed
+- Prefer `TaskGroup` for parallel loading
+- Always show a loading state while awaiting
+- Handle timeouts and errors visibly and gracefully
+
+---
+
+## File & Folder Structure
+- One view, model, and service per file
+- Place unit tests in `Tests/Features/FeatureName/` using mock services
+
+---
+
+## Testing & Mocks
+- Minimum 80% coverage for ViewModels and Services
+- Use XCTest for unit tests
```

Comment:
It's good practice to use [Swift Testing](https://developer.apple.com/xcode/swift-testing), a modern pure-Swift alternative to XCTest, to test business code that does not depend on platform stuff.

Reply:
I want to comment that we [don't support it yet in KMT](https://youtrack.jetbrains.com/issue/KMT-1057/Swift.-Swift-Testing-Support-for-Test-Pane)